src/integration/java/io/pinecone/integration/controlPlane/serverless/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -1,52 +1,131 @@
-        controlPlaneClient = new Pinecone(System.getenv("PINECONE_API_KEY"));
-    }
-
-    @Test
-    public void createAndDelete() throws InterruptedException {
-        String indexName = RandomStringBuilder.build("index-name", 8);
+    private static Pinecone controlPlaneClient = new Pinecone(System.getenv("PINECONE_API_KEY"));
+    private static final String indexName = RandomStringBuilder.build("create-index", 8);
+    @BeforeAll
+    public static void setUp() throws InterruptedException {
+        // Create the index
         ServerlessSpec serverlessSpec = new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AWS).region("us-west-2");
         CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().serverless(serverlessSpec);
-
-        // Create the index
         CreateIndexRequest createIndexRequest = new CreateIndexRequest()
                 .name(indexName)
                 .metric(IndexMetric.COSINE)
                 .dimension(10)
                 .spec(createIndexRequestSpec);
         controlPlaneClient.createIndex(createIndexRequest);
-
         waitUntilIndexIsReady(controlPlaneClient, indexName);
+    }
 
+    @AfterAll
+    public static void cleanUp() {
+        // Delete the index
+        controlPlaneClient.deleteIndex(indexName);
+    }
+
+    @Test
+    public void describeAndListIndex() {
         // Describe the index
         IndexModel indexModel = controlPlaneClient.describeIndex(indexName);
         assertNotNull(indexModel);
         assertEquals(10, indexModel.getDimension());
         assertEquals(indexName, indexModel.getName());
         assertEquals(IndexMetric.COSINE, indexModel.getMetric());
+        assertNotNull(indexModel.getSpec().getServerless());
 
         // List the index
         IndexList indexList = controlPlaneClient.listIndexes();
         assertNotNull(indexList.getIndexes());
+        assertTrue(indexList.getIndexes().stream().anyMatch(index -> indexName.equals(index.getName())));
+    }
 
-        // Delete the index
-        controlPlaneClient.deleteIndex(indexName);
-        Thread.sleep(3500);
+    @Test
+    public void createIndexWithInvalidName() {
+        ServerlessSpec serverlessSpec = new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AWS).region("us-west-2");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().serverless(serverlessSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("Invalid-name")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("must consist of lower case alphanumeric characters or '-'"));
```

Comment:
Total nit, but technically shouldn't the "or" here be "and/or"? Currently to me it reads like the name can consist of lowercase alphanumeric chars OR just a hyphen, which isn't correct (I mean, technically you could do that I guess?)